simulator/ssh-transfer.go

```diff
@@ -0,0 +1,158 @@
+package simulator
+
+import (
+	"context"
+	"fmt"
+	"math/rand"
+	"net"
+	"os"
+	"strconv"
+	"strings"
+	"time"
+
+	simssh "github.com/alphasoc/flightsim/simulator/ssh"
+
+	bytesize "github.com/inhies/go-bytesize"
+)
+
+const defaultSendSize = 100 * bytesize.MB
+
+// TODO where will the server side live?
+var defaultTargetHosts = []string{"127.0.0.1:22", "127.0.0.1:9999"}
+
+// SSHTransfer defines this simulation.
+type SSHTransfer struct {
+	src      net.IP // Connect from this IP.
+	sendSize bytesize.ByteSize
+}
+
+// NewSSHTransfer creates a new SSH/SFTP simulator.
+func NewSSHTransfer() *SSHTransfer {
+	return &SSHTransfer{}
+}
+
+// Init sets the source IP for this simulation.
+func (s *SSHTransfer) Init(src net.IP) error {
+	s.src = src
+	return nil
+}
+
+// writeRandom writes sendSize bytes of 'random' data to the server.  An error is returned.
+func writeRandom(c *simssh.Client, handle string, sendSize bytesize.ByteSize) error {
+	// 8K writes.
+	const buffSize uint64 = 8192
+	toSendStr := sendSize.Format("%.0f", "B", false)
+	toSend, err := strconv.ParseUint(toSendStr[:len(toSendStr)-1], 10, 64)
+	if err != nil {
+		return fmt.Errorf("failed write: %v", err)
+	}
+	// Seed once.
+	rand.Seed(time.Now().UnixNano())
```

Comment:
rand.Seed is a global function, so it's not a good idea to call it here as it may have side-effect somewhere else. As a rule of thumb `rand.Seed` should be called somewhere in main(), once per app. You can also create own `rand.Rand` object and seed it here – it won't have any side effects and will be mutex-free – that's probably the cleanest option.

Reply:
Understood.  I've created my own rand.Rand from within the simulation Init().